lstchain/irf/tests/test_hdus.py

```diff
@@ -0,0 +1,64 @@
+import os
+import pytest
+import pandas as pd
+from astropy.io import fits
+from astropy.table import Table
+
+from lstchain.io.io import dl2_params_lstcam_key, read_data_dl2_to_QTable
+
+
+@pytest.mark.private_data
+@pytest.mark.run(after="test_write_dl2_dataframe")
+def test_create_event_list(simulated_dl2_file, simulated_irf_file):
+    from lstchain.irf.hdu_table import create_event_list
+    import numpy as np
+
+    dl2_file_new = simulated_dl2_file.parent / simulated_dl2_file.name.replace(
+        ".h5", "_new.h5"
+    )
+
+    dl2 = pd.read_hdf(simulated_dl2_file, key=dl2_params_lstcam_key)
+
+    # Adding some necessary columns for reading it as real data file
+    # Simulated data file is being used as this is run before the test_lstchain_scripts
+    dl2["tel_id"] = dl2["tel_id"].min()
+    dl2["dragon_time"] = dl2["tel_id"] + np.arange(0, len(dl2["tel_id"]) * 1e-3, 1e-3)
+    dl2["alt_tel"] = dl2["mc_alt_tel"]
+    dl2["az_tel"] = dl2["mc_az_tel"]
+    dl2.to_hdf(dl2_file_new, key=dl2_params_lstcam_key)
+
+    events = read_data_dl2_to_QTable(dl2_file_new)
+    evts, gti, pnt = create_event_list(events, run_number=0, source_name="Crab")
+
+    assert "TIME" in Table.read(evts).columns
+    assert "START" in Table.read(gti).columns
+    assert "RA_PNT" in Table.read(pnt).meta
+
+    dl3_file = dl2_file_new.name.replace("dl2", "dl3")
+    dl3_file = simulated_dl2_file.parent / dl3_file.replace(".h5", ".fits")
+    # create a temp dl3 file to test indexing function
+    edisp = fits.open(simulated_irf_file)["ENERGY DISPERSION"]
+    aeff = fits.open(simulated_irf_file)["EFFECTIVE AREA"]
+    bkg = fits.open(simulated_irf_file)["BACKGROUND"]
+    psf = fits.open(simulated_irf_file)["PSF"]
+
+    temp_hdulist = fits.HDUList(
+        [fits.PrimaryHDU(), evts, gti, pnt, aeff, edisp, bkg, psf]
+    )
+    temp_hdulist.writeto(dl3_file, overwrite=True)
+
+    assert dl3_file.is_file()
+
+
+@pytest.mark.private_data
+@pytest.mark.run(after="test_create_event_list")
+def test_create_obs_hdu_index(simulated_dl2_file):
```

Comment:
I guess the private_data flag is not needed here, right?. This is for simulated data.

Reply:
Yes! Missed removing it while I was checking the tests. Will remove it from this test file.